engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheHashingJobActor.scala

```diff
@@ -0,0 +1,267 @@
+
+    val runtimeAttributeHashes = runtimeAttributeDefinitions map { definition => jobDescriptor.runtimeAttributes.get(definition.name) match {
+      case Some(wdlValue) => HashResult(HashKey("runtime attribute: " + definition.name, definition.usedInCallCaching), wdlValue.valueString.md5HashValue)
+      case None => HashResult(HashKey("runtime attribute: " + definition.name, definition.usedInCallCaching), UnspecifiedRuntimeAttributeHashValue)
+    }}
+
+    val inputHashResults = nonFileInputs map {
+      case WdlValueSimpleton(name, value) => HashResult(HashKey(s"input: ${value.wdlType.toWdlString} $name"),  value.toWdlString.md5HashValue)
+    }
+
+    val outputExpressionHashResults = jobDescriptor.call.task.outputs map { output =>
+      HashResult(HashKey(s"output expression: ${output.wdlType.toWdlString} ${output.unqualifiedName}"), output.requiredExpression.valueString.md5HashValue)
+    }
+
+    // Build these all together for the final set of initial hashes:
+    Set(commandTemplateHash, backendNameHash, inputCountHash, outputCountHash) ++ runtimeAttributeHashes ++ inputHashResults ++ outputExpressionHashResults
+  }
+
+  private [callcaching] def makeFileHashingActor() = {
+    val fileHashingActorName = s"FileHashingActor_for_${jobDescriptor.key.tag}"
+    context.actorOf(fileHashingActorProps, fileHashingActorName)
+  }
+}
+
+object CallCacheHashingJobActor {
+
+  def props(jobDescriptor: BackendJobDescriptor,
+            callCacheReadingJobActor: Option[ActorRef],
+            initializationData: Option[BackendInitializationData],
+            runtimeAttributeDefinitions: Set[RuntimeAttributeDefinition],
+            backendName: String,
+            fileHashingActorProps: Props,
+            writeToCache: Boolean
+           ) = Props(new CallCacheHashingJobActor(
+    jobDescriptor,
+    callCacheReadingJobActor,
+    initializationData,
+    runtimeAttributeDefinitions,
+    backendName,
+    fileHashingActorProps,
+    writeToCache
+  ))
+
+  sealed trait CallCacheHashingJobActorState
+  case object WaitingForHashFileRequest extends CallCacheHashingJobActorState
+  case object HashingFiles extends CallCacheHashingJobActorState
+
+  /**
+    * Aggregates hash results together in a predictable and reproducible manner.
+    * Purposefully takes an Iterable, because it will be sorted appropriately in this method
+    * to ensure deterministic result, so the type of collection doesn't matter.
+    * Only aggregates hash values for which checkForHitOrMiss is true.
+    * If several hash keys are identical, the result of this method is undefined.
+    */
+  private def calculateHashAggregation(hashes: Iterable[HashResult], messageDigest: MessageDigest) = {
+    val sortedHashes = hashes.toList.filter(_.hashKey.checkForHitOrMiss).sortBy(_.hashKey.key).map(_.hashValue.value.getBytes)
+    sortedHashes foreach messageDigest.update
+    DatatypeConverter.printHexBinary(messageDigest.digest())
```

Comment:
Yes at least to the extent of the testing I've done they yield the same.
I ran this on the 10K mock wdl database after the migration and it call cached immediately.
`sortedHashes foreach messageDigest.update` is equivalent to md5ing the concatenation.
`DatatypeConverter.printHexBinary` just prints the bytes in hexadecimal string format.